src/main/java/org/candlepin/model/ContentQueryBuilder.java

```diff
@@ -0,0 +1,666 @@
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.model;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.inject.Provider;
+import javax.persistence.Column;
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import javax.persistence.metamodel.EntityType;
+
+
+
+/**
+ * The ContentQueryBuilder is a utility class which provides a fluent-style interface for performing
+ * arbitrary content lookups with a number of parameters.
+ * <p></p>
+ * This class is a loose wrapper around the JPA interface and one or more native queries to fetch results
+ * as efficiently as possible, without the caller needing to jump back and forth between the curators
+ * to get both the count and results.
+ */
+public class ContentQueryBuilder extends QueryBuilder<ContentQueryBuilder, Content> {
+    private static final Logger log = LoggerFactory.getLogger(ContentQueryBuilder.class);
+
+    private final Set<String> contentIds;
+    private final Set<String> contentLabels;
+    private final Map<String, Owner> owners;
+    private Inclusion active;
+    private Inclusion custom;
+
+    /**
+     * Creates a new query builder backed by the specified entity manager provider.
+     * <p></p>
+     * <strong>Note:</strong> Query builder instances should not be constructed directly and should be
+     * fetched from their corresponding curators.
+     *
+     * @param entityManagerProvider
+     *  the provider to use for fetching an entity manager when building queries
+     */
+    protected ContentQueryBuilder(Provider<EntityManager> entityManagerProvider) {
+        super(entityManagerProvider);
+
+        this.contentIds = new HashSet<>();
+        this.contentLabels = new HashSet<>();
+        this.owners = new HashMap<>();
+        this.active = Inclusion.INCLUDE;
+        this.custom = Inclusion.INCLUDE;
+    }
+
+    /**
+     * Adds one or more content IDs to use for filtering query output.
+     *
+     * @param contentIds
+     *  a collection of content IDs to use to filter query output, or null to clear any previously
+     *  set content IDs
+     *
+     * @return
+     *  a reference to this query builder
+     */
+    public ContentQueryBuilder addContentIds(Collection<String> contentIds) {
+        if (contentIds == null) {
+            return this;
+        }
+
+        contentIds.stream()
+            .filter(pid -> pid != null && !pid.isBlank())
```

Comment:
Update from `pid` to `cid` for content ID

Reply:
Fixed